src/biome/text/tokenizer.py

```diff
@@ -109,48 +99,50 @@ def _tokenize(self, text: str) -> List[Token]:
         tokens: `List[Token]`
 
         """
-        return self._base_tokenizer.tokenize(text[: self.max_sequence_length])
+        tokens = self.nlp(text[: self.max_sequence_length])
+        return self._sanitize(_remove_spaces(tokens) if self._remove_spaces else tokens)
 
     def tokenize_document(self, document: List[str]) -> List[List[Token]]:
         """ Tokenizes a document-like structure containing lists of text inputs
 
-        Use this to account for hierarchical text structures (e.g., a paragraph is a list of sentences)
+        Use this to account for hierarchical text structures (e.g., a paragraph)
 
         Parameters
         ----------
         document: `List[str]`
-            A `List` with text inputs, e.g., sentences
-            
+            A `List` with text inputs, e.g., paragraphs
+
         Returns
         -------
         tokens: `List[List[Token]]`
         """
-        sentences = [self.text_cleaning(text) for text in document]
+        paragraphs = [self.text_cleaning(text) for text in document]
         if self.segment_sentences:
-            sentences = [
-                sentence
-                for sentences in self.segment_sentences.batch_split_sentences(sentences)
-                for sentence in sentences
+            paragraphs = [
+                sentence.string.strip()
+                for doc in self.__nlp__.pipe(paragraphs)
+                for sentence in doc.sents
             ]
+
         return [
-            self._tokenize(sentence)
-            for sentence in sentences[: self.max_nr_of_sentences]
+            self._tokenize(paragraph)
+            for paragraph in paragraphs[: self.max_nr_of_sentences]
```

Comment:
The name `self.max_nr_of_sentences` can be a bit confusing here, since if `self.segment_sentences` is False, then `self.max_nr_of_sentences` will refer to the number of paragraphs.

Reply:
Agree. Maybe limit a number of sentences make sense when `segment_sentences` is enabled. What do you think?

Reply:
yes, like i wrote below, i am not sure if the `segment_sentences` option makes sense for a "list of text" input.